tests/e2e/support/page/files/publicLink.ts

```diff
@@ -0,0 +1,227 @@
+    this.expirationDateDropdownLocator = this.actor.page.locator('#files-links-expiration-btn')
+    this.createLinkButtonLocator = this.actor.page.locator('#oc-files-file-link-create')
+    this.publicLinkLocator = this.actor.page.locator(
+      '//a[@class = "oc-files-file-link-url oc-text-truncate"]'
+    )
+    this.yearButtonLocator = this.actor.page.locator(
+      `//div[@class = "vc-nav-container"]/div[@class="vc-nav-header"]//span[position()=2]`
+    )
+    this.nextSpanYearLocator = this.actor.page.locator(
+      `//div[@class = "vc-nav-container"]/div[@class="vc-nav-header"]//span[position()=3]`
+    )
+    this.monthAndYearDropdownLocator = this.actor.page.locator(`//div[@class = 'vc-title']`)
+    // for chaning values of selector the below selectors are made string for passing value
+    this.daySelector = `//span[@tabindex='-1' or @tabindex='0'][@aria-label='%s']`
+    this.monthSelector = `//span[@data-id='%s.%s']`
+    this.yearSelector = `//div[@class = "vc-nav-container"]/div[@class="vc-nav-items"]//span[contains(text(),'%s')]`
+    this.publicLinkListSelector = `//ul[@class = 'oc-list oc-list-divider oc-overflow-hidden oc-m-rm']/li`
+    this.roleSelector = `//span[@id="files-role-%s"]`
+    this.folderSelector = `//*[@data-test-resource-name="%s"]/ancestor::tr//button[contains(@class, "files-quick-action-collaborators")]`
+  }
+
+  async selectRole(role: string): Promise<void> {
+    await this.roleDropdownLocator.click()
+    await this.actor.page.locator(util.format(this.roleSelector, role)).click()
+  }
+
+  async selectExpiryMonth(year: string, month: string): Promise<void> {
+    await this.actor.page.locator(util.format(this.monthSelector, year, month)).click()
+  }
+
+  async selectExpiryDay(dayMonthYear: string): Promise<void> {
+    await this.actor.page.locator(util.format(this.daySelector, dayMonthYear)).click()
+  }
+
+  getDateType = (expiryDate: string): string => {
+    if (expiryDate.charAt(0).includes('-')) {
+      throw new Error('The provided date is negative and has already expired !!')
+    } else if (expiryDate.charAt(0).includes('+')) {
+      return expiryDate.toLowerCase().match(/[dayrmonthwek]+/)[0]
+    }
+  }
+
+  addMonth = (noOfMonths: number): Date => {
+    const date = new Date()
+    date.setMonth(date.getMonth() + noOfMonths)
+    return date
+  }
+
+  addDays = (days: number): Date => {
+    const date = new Date()
+    date.setDate(date.getDate() + days)
+    return date
+  }
+
+  setActualExpiryDate = (dateType: string, dateOfExpiration: string): Date => {
+    switch (dateType) {
+      case 'day':
+        return this.addDays(parseInt(dateOfExpiration))
+      case 'week':
+        return this.addDays(parseInt(dateOfExpiration) * 7)
+      case 'month':
+        return this.addMonth(parseInt(dateOfExpiration))
+      case 'year':
+        return new Date(
+          new Date().setFullYear(new Date().getFullYear() + parseInt(dateOfExpiration))
+        )
+    }
+  }
```

Comment:
helper functions? if not related to this PageObject, can be referenced from specific helpers.